lib/Sema/ConstraintSystem/LocalCSWalker.cpp

```diff
@@ -228,9 +229,77 @@ class LocalCSWalker : public glu::ast::ASTWalker<LocalCSWalker, void> {
         _cs.addConstraint(constraint);
     }
 
-    void _visitForStmt([[maybe_unused]] glu::ast::ForStmt *node)
+    void _visitForStmt(glu::ast::ForStmt *node)
     {
-        // TODO: #439 Implement for-loops
+        auto *binding = node->getBinding();
+        auto *range = node->getRange();
+
```

Comment:
you need to set the root of the CS in such a way that it doesn't include the body of the for loop, similar to the while loop (it needs to go over the rest of the for loop content, however
Or, you can override _visitForStmt in `TypeVariableTyMapper` so it doesn't go over the body
Or, fix the literal expressions to have a null type instead of type variables, just like other expressions.
Make sure this code compiles:
```
for i in 0..<10 {
 let j = i + 1;
}
```